### DISCUSSION TITLE
Running apt-offline but no updates happening

### DISCUSSION THREAD

**Discussion by mcarvalho-astranis** [rickysarraf/apt-offline#223] [Q&A]
Hi Ricky

Thanks for this great package.

Have following instruction on offline upgrade from:  

          https://www.debugpoint.com/how-to-update-or-upgrade-ubuntu-offline-without-internet/

Checked apt list --upgradable (on remote host) and shows 26 packages to upgrade.

Have the apt-offline.sig file on the remote host created.

Minor change was done on "set" command:  (added --update --upgrade to the line)

         # apt-offline set ./offline-data.sig --update --upgrade

Move to connected computer run "get" command

apt-offline get /tmp/offline-data.sig --bug-reports --threads 5 -d /tmp/Offline-Get/ > /tmp/offline.out    2> /tmp/offline.errors

Issue:  

1) On running get command:  (at the connected computer)
I get a lot (235) of correct hits "Downloading http:...."I get too many (105) "ERROR: Giving up on URL...."
I think 105 errors among 340 tries is too many.

Question(1): Can please anyone chip in and let me know if this is expected, or if you see what I possibly doing incorrectly?

Brought Offline-Get/  to remote computer.

Ran apt-offline install

        # apt-offline install Offline-GET/

We see "Proceeding with installation....What would you like to do next: (y, N, ?) Y.....

Exit status is "zero," but we only see three packages returning "synced" and the proces stops 

issue: 

2) No update has happened on any of the 26 packages listed as upgradable, although they are seen on the "Downloading" output of the "get" command.

Question(2):  What am I doing incorrect?

Please advise.

Many thanks

Marcelo

**Comment by rickysarraf** [comment]
Could you please share the full CLI output, along with all the logs ? That'll help ascertain what the problem is.

**Reply by rickysarraf** [reply]
>On running get command: (at the connected computer). I get a lot (235) of correct hits "Downloading http:...."I get too many (105) "ERROR: Giving up on URL...."
I think 105 errors among 340 tries is too many.



Those errors should have been fine. apt-offline cycles through supported archive formats on the remote server, before ultimately giving up. Full console log would have confirmed the same.


See below code:

https://github.com/rickysarraf/apt-offline/blob/a6cc640d4222d0d0af374801db25029dc5636d55/apt_offline_core/AptOfflineCoreLib.py#L1396-L1417



I think your problem is the fact that you are running the `set` command with `--update and --upgrade` together. While it works, and was the default previously, using those options together is highly discouraged.

Discouraged so much, that I changed those default. See this: 2cab46a31b65c1d4831da3b0b0ae0d7a2e4004dd



To sum up, I'd encourage you to check out the wiki documentation that has been prepared.
* https://github.com/rickysarraf/apt-offline/wiki

**Reply by domo141** [reply]
the wiki is still a bit outdated -- in the first place where `apt-offline set file.sig` is mentioned no options (e.g. `--update`) is there

[EDIT: the exact form is `apt-offline set /tmp/apt-offline.txt` -- with `.sig` that would be more consistent :)]

I read very recently updated page somewhere and same thing was there -- it is somewhat confusing to get error message "no options there" when trying on someone's example...

That said, thanks for the great tool, it seems to work pretty well..!

**Comment by mcarvalho-astranis** [comment]
Hi Ricky

Many thanks for opening this discussion.

My apologies for not return yesterday.  I was scripting my own solution.  It worked but I prefer to run with a "institutionalized" package then my own scripts for obvious reason.

I will proceed with "share the full CLI output, along with all the logs."  Please give e a couple of hours.

Are there specific logs location I need to be aware off?

Please advise.

Acton Item:  From me:  "share the full CLI output, along with all the logs."

**Comment by mcarvalho-astranis** [comment]
Hi Ricky

Here we go.

Have following instruction on offline upgrade from:

      https://www.debugpoint.com/how-to-update-or-upgrade-ubuntu-offline-without-internet/

I have tried twice.  First on a Intel NUC that was online but I faked it offline.  Secondly on an offline DEV host that mirrors my production one.   By chance I lost the Pen Drive, it died, and I have no logs from the DEV host just the CLI commands and answers.  Please bear with me.

           # Creating Signature file - at the offline host
# apt-offline set ./sf-moc-changeme_offline-data.sig --update --upgrade
    Gathering details needed for 'update' operation
    Gathering details needed for 'upgrade' operation
# echo $?
    0

            # Moving Sig file to Connected host.  I kept the sig file on the Pen Drive and created a directory named the offline host
# mkdir sf-moc-changeme
# mv sf-moc-changeme_offline-data.sig sf-moc-changeme/
# cd sf-moc-changeme/
# ls
    Sf-moc-changeme_offline-data.sig

               # Creating “Get” directory
# mkdir Offline-GET_sf-moc-changeme

             # Getting required packages
# apt-offline get ./sf-moc-changeme_offline-data.sig --bug-reports --threads 5 -d ./Offline-GET_sf-moc-changeme/  > offline.out    2> offline.errors
# ls
    offline.errors  Offline-GET_sf-moc-changeme  offline.out  sf-moc-changeme_offline-data.sig

Attention:  here is were I had my offline.out and offline.errors files lost when the Pen Drive died.  No harm to the actions that follows

            # Moving “Get” directory to offline mirror


            # Install the downloaded packages to the offline system.
                   Pre-install
# mkdir Offline-GET
# ls /media/xxxxxxxxxxx/Samsung\ USB/MOC/sf-moc-changeme/Offline-GET_sf-moc-changeme/ | wc -l
    138                  # This is data from the Pen Drive
# ls Offline-GET/ | wc -l
    0                      # This is local to the offline host
# cp /media/xxxxxxxxxxx/Samsung\ USB/MOC/sf-moc-changeme/Offline-GET_sf-moc-changeme/* Offline-GET/
# ls Offline-GET/ | wc -l
    138

            # Install the downloaded packages to the offline system
                         Sanity check
# apt update
          -----
    26 packages can be upgraded. Run 'apt list --upgradable' to see them.

                    Go for offline update installs
# apt-offline install Offline-GET/
    Proceeding with installation
           --------
    Following are the list of bugs present.
    1051028 google-chrome-stable : google-chrome-stable: Google Chrome does not respect prefers-color-scheme for webpages on KDE Plasma
    (Y) Yes. Proceed with installation
    (N) No, Abort.
    (R) Redisplay the list of bugs.
    (Bug Number) Display the bug report from the Offline Bug Reports.
    Display this help message.
    What would you like to do next: (y, N, ?)Y
    libsnmp35_5.8+dfsg-2ubuntu2.6_amd64.deb file synced.
    google-chrome-stable_109.0.5414.74-1_amd64.deb file synced.
    libsnmp-base_5.8+dfsg-2ubuntu2.6_all.deb file synced. 
# echo $?
    0

         # Validating
# apt update
        -----
    Reading package lists... Done
    Building dependency tree       
    Reading state information... Done
    26 packages can be upgraded. Run 'apt list --upgradable' to see them.

               Did not work, 26 packages yet to upgrade.

Note:  I am attaching .out and .errors files here that ARE NOT from this test as we lost it on the Pen Drive.  They are from the first test I did on the NUC and also did not work.

Please advise if you see I am missing something.

One last thing.  Is there any logs you can point our to me to take a look?


Many thanks


Marcelo

[offline.errors.TXT](https://github.com/rickysarraf/apt-offline/files/12807138/offline.errors.TXT)
[offline.out.TXT](https://github.com/rickysarraf/apt-offline/files/12807139/offline.out.TXT)

**Reply by rickysarraf** [reply]
```
[1;0m[1;31mERROR: Giving up on URL http://us.archive.ubuntu.com/ubuntu/dists/focal-backports/main/i18n/Translation-en_US.lzma                                                             
[1;0m[1;32m[1;0m[1;32m[1;0m[1;32m[1;0m[1;31mERROR: Giving up on URL http://us.archive.ubuntu.com/ubuntu/dists/focal-backports/main/dep11/Components-all.yml.lzma                                                             
[1;0m[1;32m[1;0m[1;32m[1;0m[1;32m[1;0m[1;32m[1;0m[1;31mERROR: Giving up on URL http://us.archive.ubuntu.com/ubuntu/dists/focal-backports/main/cnf/Commands-all.lzma                                                             
[1;0m[1;31mERROR: Giving up on URL http://us.archive.ubuntu.com/ubuntu/dists/focal-backports/restricted/binary-all/Packages.lzma                                                             
[1;0m[1;31mERROR: Giving up on URL http://us.archive.ubuntu.com/ubuntu/dists/focal-backports/restricted/i18n/Translation-en_US.lzma                                                             
[1;0m[1;31mERROR: Giving up on URL http://us.archive.ubuntu.com/ubuntu/dists/focal-backports/restricted/dep11/Components-amd64.yml.lzma                                                             
[1;0m[1;31mERROR: Giving up on URL http://us.archive.ubuntu.com/ubuntu/dists/focal-backports/restricted/dep11/Components-all.yml.lzma                                                             
[1;0m[1;32m[1;0m[1;31mERROR: Giving up on URL http://us.archive.ubuntu.com/ubuntu/dists/focal-backports/restricted/dep11/icons-48x48.tar.lzma                                                             
[1;0m[1;32m[1;0m[1;32m[1;0m[1;31mERROR: Giving up on URL http://us.archive.ubuntu.com/ubuntu/dists/focal-backports/restricted/dep11/icons-64x64@2.tar.lzma                                                             
[1;0m[1;31mERROR: Giving up on URL http://us.archive.ubuntu.com/ubuntu/dists/focal-backports/restricted/dep11/icons-64x64.tar.lzma                                                             
[1;0m[1;31mERROR: Giving up on URL http://us.archive.ubuntu.com/ubuntu/dists/focal-backports/restricted/cnf/Commands-all.lzma                                                             
[1;0m[1;32m[1;0m[1;32m[1;0m[1;31mERROR: Giving up on URL http://us.archive.ubuntu.com/ubuntu/dists/focal-backports/universe/binary-all/Packages.lzma                                                             
[1;0m[1;31mERROR: Giving up on URL http://us.archive.ubuntu.com/ubuntu/dists/focal-backports/universe/i18n/Translation-en_US.lzma                                                             
[1;0m[1;32m[1;0m[1;32m[1;0m[1;31mERROR: Giving up on URL http://us.archive.ubuntu.com/ubuntu/dists/focal-backports/universe/dep11/Components-all.yml.lzma                                                             
[1;0m[1;32m[1;0m[1;32m[1;0m[1;32m[1;0m[1;32m[1;0m[1;32m[1;0m[1;31mERROR: Giving up on URL http://us.archive.ubuntu.com/ubuntu/dists/focal-backports/universe/cnf/Commands-all.lzma                                                             
[1;0m[1;31mERROR: Giving up on URL http://us.archive.ubuntu.com/ubuntu/dists/focal-backports/multiverse/binary-all/Packages.lzma                                                             
[1;0m[1;31mERROR: Giving up on URL http://us.archive.ubuntu.com/ubuntu/dists/focal-backports/multiverse/i18n/Translation-en_US.lzma                                                             
[1;0m[1;31mERROR: Giving up on URL http://us.archive.ubuntu.com/ubuntu/dists/focal-backports/multiverse/dep11/Components-amd64.yml.lzma                                                             
[1;0m[1;31mERROR: Giving up on URL http://us.archive.ubuntu.com/ubuntu/dists/focal-backports/multiverse/dep11/Components-all.yml.lzma                                                             
[1;0m[1;32m[1;0m[1;31mERROR: Giving up on URL http://us.archive.ubuntu.com/ubuntu/dists/focal-backports/multiverse/dep11/icons-48x48.tar.lzma                                                             
[1;0m[1;31mERROR: Giving up on URL http://us.archive.ubuntu.com/ubuntu/dists/focal-backports/multiverse/dep11/icons-64x64.tar.lzma                                                             
[1;0m[1;31mERROR: Giving up on URL http://us.archive.ubuntu.com/ubuntu/dists/focal-backports/multiverse/cnf/Commands-all.lzma                                                             
[1;0m[1;31mERROR: Giving up on URL http://us.archive.ubuntu.com/ubuntu/dists/focal-backports/multiverse/dep11/icons-64x64@2.tar.lzma                                                             
[1;0m[1;32m[1;0mCalling get_bugs with positional arguments is deprecated, please use keyword arguments instead.
[1;32m[1;0m[1;31mERROR: Some items failed to download. Downloaded data may be incomplete
[1;0m[1;31mERROR: Please run in verbose mode to see details about failed items
```

These errors are fine. They are just complained about the missing metadata payload, which isn't critical to functionality

**Comment by mcarvalho-astranis** [comment]
Ricky

I am sorry it got large fonts.  It was not me.  It was the "#" prompt I had in front of command lines that provoked that.

Marcelo

**Comment by mcarvalho-astranis** [comment]
Hi Ricky

Thank you so much.

I will go over it the next couple of days and get back here with updates.

Many thanks.

On Thu, Oct 5, 2023 at 2:00 AM Ritesh Raj Sarraf ***@***.***>
wrote:

> �[1;0m�[1;31mERROR: Giving up on URL http://us.archive.ubuntu.com/ubuntu/dists/focal-backports/main/i18n/Translation-en_US.lzma
> �[1;0m�[1;32m�[1;0m�[1;32m�[1;0m�[1;32m�[1;0m�[1;31mERROR: Giving up on URL http://us.archive.ubuntu.com/ubuntu/dists/focal-backports/main/dep11/Components-all.yml.lzma
> �[1;0m�[1;32m�[1;0m�[1;32m�[1;0m�[1;32m�[1;0m�[1;32m�[1;0m�[1;31mERROR: Giving up on URL http://us.archive.ubuntu.com/ubuntu/dists/focal-backports/main/cnf/Commands-all.lzma
> �[1;0m�[1;31mERROR: Giving up on URL http://us.archive.ubuntu.com/ubuntu/dists/focal-backports/restricted/binary-all/Packages.lzma
> �[1;0m�[1;31mERROR: Giving up on URL http://us.archive.ubuntu.com/ubuntu/dists/focal-backports/restricted/i18n/Translation-en_US.lzma
> �[1;0m�[1;31mERROR: Giving up on URL http://us.archive.ubuntu.com/ubuntu/dists/focal-backports/restricted/dep11/Components-amd64.yml.lzma
> �[1;0m�[1;31mERROR: Giving up on URL http://us.archive.ubuntu.com/ubuntu/dists/focal-backports/restricted/dep11/Components-all.yml.lzma
> �[1;0m�[1;32m�[1;0m�[1;31mERROR: Giving up on URL http://us.archive.ubuntu.com/ubuntu/dists/focal-backports/restricted/dep11/icons-48x48.tar.lzma
> �[1;0m�[1;32m�[1;0m�[1;32m�[1;0m�[1;31mERROR: Giving up on URL ***@***.***
> �[1;0m�[1;31mERROR: Giving up on URL http://us.archive.ubuntu.com/ubuntu/dists/focal-backports/restricted/dep11/icons-64x64.tar.lzma
> �[1;0m�[1;31mERROR: Giving up on URL http://us.archive.ubuntu.com/ubuntu/dists/focal-backports/restricted/cnf/Commands-all.lzma
> �[1;0m�[1;32m�[1;0m�[1;32m�[1;0m�[1;31mERROR: Giving up on URL http://us.archive.ubuntu.com/ubuntu/dists/focal-backports/universe/binary-all/Packages.lzma
> �[1;0m�[1;31mERROR: Giving up on URL http://us.archive.ubuntu.com/ubuntu/dists/focal-backports/universe/i18n/Translation-en_US.lzma
> �[1;0m�[1;32m�[1;0m�[1;32m�[1;0m�[1;31mERROR: Giving up on URL http://us.archive.ubuntu.com/ubuntu/dists/focal-backports/universe/dep11/Components-all.yml.lzma
> �[1;0m�[1;32m�[1;0m�[1;32m�[1;0m�[1;32m�[1;0m�[1;32m�[1;0m�[1;32m�[1;0m�[1;31mERROR: Giving up on URL http://us.archive.ubuntu.com/ubuntu/dists/focal-backports/universe/cnf/Commands-all.lzma
> �[1;0m�[1;31mERROR: Giving up on URL http://us.archive.ubuntu.com/ubuntu/dists/focal-backports/multiverse/binary-all/Packages.lzma
> �[1;0m�[1;31mERROR: Giving up on URL http://us.archive.ubuntu.com/ubuntu/dists/focal-backports/multiverse/i18n/Translation-en_US.lzma
> �[1;0m�[1;31mERROR: Giving up on URL http://us.archive.ubuntu.com/ubuntu/dists/focal-backports/multiverse/dep11/Components-amd64.yml.lzma
> �[1;0m�[1;31mERROR: Giving up on URL http://us.archive.ubuntu.com/ubuntu/dists/focal-backports/multiverse/dep11/Components-all.yml.lzma
> �[1;0m�[1;32m�[1;0m�[1;31mERROR: Giving up on URL http://us.archive.ubuntu.com/ubuntu/dists/focal-backports/multiverse/dep11/icons-48x48.tar.lzma
> �[1;0m�[1;31mERROR: Giving up on URL http://us.archive.ubuntu.com/ubuntu/dists/focal-backports/multiverse/dep11/icons-64x64.tar.lzma
> �[1;0m�[1;31mERROR: Giving up on URL http://us.archive.ubuntu.com/ubuntu/dists/focal-backports/multiverse/cnf/Commands-all.lzma
> �[1;0m�[1;31mERROR: Giving up on URL ***@***.***
> �[1;0m�[1;32m�[1;0mCalling get_bugs with positional arguments is deprecated, please use keyword arguments instead.
> �[1;32m�[1;0m�[1;31mERROR: Some items failed to download. Downloaded data may be incomplete
> �[1;0m�[1;31mERROR: Please run in verbose mode to see details about failed items
>
> These errors are fine. They are just complained about the missing metadata
> payload, which isn't critical to functionality
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rickysarraf/apt-offline/discussions/223#discussioncomment-7195671>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BC5URAP3U7ZJ7UQEMK4JKJ3X5ZZJVAVCNFSM6AAAAAA5TCYFW2VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TCOJVGY3TC>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

-- 


________
This email and any attachments may contain Astranis confidential 
and/or proprietary information governed by a non-disclosure agreement, and 
are intended solely for the individual or entity specified by the message.
